napari/layers/_image_layer/model.py

```diff
@@ -247,16 +259,28 @@ def interpolation_index(self, interpolation_index):
     def colormap(self):
         """string or ColorMap: Colormap to use for luminance images.
         """
-        return self._node.cmap
+        return self.colormap_name, self._node.cmap
 
     @colormap.setter
     def colormap(self, colormap):
-        try:
-            cmap = self._colormaps[colormap]
-        except KeyError:
-            pass
-
-        self._node.cmap = cmap
+        name = '[unnamed colormap]'
+        if isinstance(colormap, str):
+            name = colormap
+        elif isinstance(colormap, tuple):
+            name, cmap = colormap
+            self._colormaps[name] = cmap
+        elif isinstance(colormap, dict):
+            self._colormaps.update(colormap)
+            name = list(colormap)[0]  # first key in dict
```

Comment:
This allows someone to pass in a `dict` with multiple colormaps. Is that a workflow we want to allow? Why?

Reply:
I'm not sure why we would need multiple colormaps for one image layer

Reply:
Well, initially I thought we might want to add with dict syntax, and then I thought, maybe they want to add a bunch of colormaps to quickly try them out in the viewer. It was easy to support multiple keys. (Less easy with tuples or lists where there can be more ambiguity.)

Ultimately when we move the colormap list to a central, layer-independent location, we can revisit this.

Reply:
Well, initially I thought we might want to add with dict syntax, and then I thought, maybe they want to add a bunch of colormaps to quickly try them out in the viewer. It was easy to support multiple keys. (Less easy with tuples or lists where there can be more ambiguity.)

Ultimately when we move the colormap list to a central, layer-independent location, we can revisit this.